### ISSUE TITLE
Failed Snowflake Statement records as successfully executed

### ISSUE THREAD

**Issue by awhitford-cip** [liquibase/liquibase#4124]
### Search first

- [X] I searched and no similar issues were found

### Description

I have a changeset that fails SQL Compilation, yet Liquibase seem to think it was successful and carries on.
I see the entry in `DATABASECHANGELOG` and everything.

In one example, I am trying to add a new column like:
```sql
alter table T add column C int as (ARRAY_SIZE(E)) not null
```
Alas, this failed with the error message:
> SQL compilation error: Non-nullable column 'C' cannot be added to non-empty table 'T' unless it has a non-null default value.

The expectation is that since this statement failed to execute, that the changeset would not be recorded as successful.  The update process should have halted upon the error.

### Steps To Reproduce

The following changelog run against Snowflake should exhibit the problem:
```sql
-- liquibase formatted sql

-- changeset awhitford:create-AW_TEST
create table AW_TEST (
  id int autoincrement not null,
  e array not null
)
-- rollback drop table AW_TEST

-- changeset awhitford:insert-AW_TEST
insert into AW_TEST (e) select ARRAY_CONSTRUCT(1)
-- rollback truncate table AW_TEST

-- changeset awhitford:alter-AW_TEST
alter table AW_TEST add column c int as (ARRAY_SIZE(e)) not null
-- rollback alter table AW_TEST drop column c
```

### Expected/Desired Behavior

The expectation is that since this statement failed to execute, that the changeset would not be recorded as successful.  The update process should have halted upon the error.

### Liquibase Version

4.20.0

### Database Vendor & Version

Snowflake

### Liquibase Integration

GitHub Actions

### OS and/or Infrastructure Type/Provider

Ubuntu / GitHub Actions

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by nvoxland** [comment]
This should be fixed by #4034 which was just released as part of 4.21.0.

Can you check if 4.21.0 fixes the problem for you?

**Comment by awhitford-cip** [comment]
Anxious to try...  Waiting for the Liquibase GitHub Actions to be released.
